src/api_server.cc

```diff
@@ -200,4 +208,26 @@ void MetadataApiServer::HandleHealthz(
   }
 }
 
+void MetadataApiServer::HandleMetrics(
+    const HttpServer::request& request,
+    std::shared_ptr<HttpServer::connection> conn) {
+  std::string response = SerializeMetricsToPrometheusTextFormat();
+  conn->set_status(HttpServer::connection::ok);
+  conn->set_headers(std::map<std::string, std::string>({
+    {"Connection", "close"},
+    {"Content-Length", std::to_string(response.size())},
+    {"Content-Type", "application/json"},
+  }));
+  conn->write(response);
+}
+
+std::string MetadataApiServer::SerializeMetricsToPrometheusTextFormat() const {
```

Comment:
in the current implementation, it is implemented as function inside `google::internal` namespace, https://github.com/Stackdriver/metadata-agent/pull/210/files#diff-bcca2b17db5d9ee49a78774f858b95e4R25 . Two reasons:
1. It can be implemented as a function, instead of a class
2. implement the function inside `google::internal` namespace enables us to test this function, but does not need to keep an member `telemetry_` inside the `MetadataApiServer` class